src/components/FriendshipButton/FriendshipButton.container.ts

```diff
@@ -39,7 +39,13 @@ const mapState = (state: RootState, ownProps: OwnProps): MapStateProps => {
 const mapDispatch = (dispatch: MapDispatch, ownProps: OwnProps): MapDispatchProps => ({
   onCancelFriendRequest: () => dispatch(cancelFriendshipRequestRequest(ownProps.friendAddress)),
   onRemoveFriend: () => dispatch(removeFriendRequest(ownProps.friendAddress)),
-  onAddFriend: () => dispatch(requestFriendshipRequest(ownProps.friendAddress)),
+  onAddFriend: () => {
+    if (ownProps.isLoggedIn) {
```

Comment:
Just a reminder that if `isLoggedIn` is something that can be obtained from redux, you can always use the value from mapDispatch on the third argument (merge) from connect. Updating the dispatch with the value of the state without the need of providing the prop to the component.